### PR TITLE
3P Media: Add opt-in incremental search

### LINKED ISSUE #3206: Remove search debounce and use enter to search
As per UX feedback

### PR DESCRIPTION
**Pull request by diegovar**
## Summary

Updates searchInput with incremental search and debounce functionality. Optionally use this based on a feature flag in the upload and media3p tabs.

## Relevant Technical Choices

Decided to not wrap searchInput and instead roll the functionality into it, as it introduced unnecessary nesting and logic.

## To-do

Determine whether incremental search is desired with UX.

## User-facing changes

The upload tab and media3p tab now do not do incremental search. Instead, the user has to press enter to search (unless they've cleared the text in which case autosearch occurs automatically in that case).

## Testing Instructions

Verify searching works fine by using Enter to search.

---

Fixes #3206

### CODE REVIEW

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js [hunk 1]
```diff
@@ -298,6 +277,10 @@ function MediaPane(props) {
     [hasMore, isMediaLoading, isMediaLoaded, setNextPage]
   );
 
+  const onSearch = (v) => setSearchTerm({ searchTerm: v });
+
+  const autoSearchDebounceMedia3p = useFeature('autoSearchDebounceMedia3p');
```
Comment (swissspidy) [inline assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: A bit confusing to use a media3p flag for local media

**Comment by swissspidy** [comment]
I believe the correct term for this is [incremental search](https://en.wikipedia.org/wiki/Incremental_search). Worth renaming?

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js (hunk 1 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: Renamed the flag.

**Comment by diegovar** [comment]
> I believe the correct term for this is [incremental search](https://en.wikipedia.org/wiki/Incremental_search). Worth renaming?

I've renamed the feature to incremental.

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js [hunk 2]
```diff
@@ -298,6 +277,12 @@ function MediaPane(props) {
     [hasMore, isMediaLoading, isMediaLoaded, setNextPage]
   );
 
+  const onSearch = (v) => setSearchTerm({ searchTerm: v });
+
+  const incrementalSearchDebounceMedia = useFeature(
+    'incrementalSearchDebounceMedia'
```
Comment (winterDroid) [inline assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: For global visibility, shall we have those with a little description in a separate file as const?

File: includes/Story_Post_Type.php [hunk 3]
```diff
@@ -465,6 +465,13 @@ public function get_editor_settings() {
 				 * Creation date: 2020-06-30
 				 */
 				'rowBasedGallery'              => false,
+				/**
+				 * Description: Flag for using an as-you-type search in media and media3p with a debouncer.
+				 * Author: @diegovar
+				 * Issue: #3206
+				 * Creation date: 2020-07-15
+				 */
+				'autoSearchDebounceMedia'      => false,
```
Comment (winterDroid) [inline includes/Story_Post_Type.php]: Shouldn't this be `incrementalSearchDebounceMedia`? 🤔

**Review by winterDroid — APPROVED** [review]

File: assets/src/edit-story/components/library/panes/media/local/mediaPane.js (hunk 2 above)
Comment (diegovar) [reply assets/src/edit-story/components/library/panes/media/local/mediaPane.js]: added to flags.js

File: includes/Story_Post_Type.php (hunk 3 above)
Comment (diegovar) [reply includes/Story_Post_Type.php]: yes indeed!

**Pull request merged**